### PR TITLE
[ADD] Add pos_analytic_config module

### PR DESCRIPTION
**Pull request by adrienpeiffer**
Use analytic account defined on POS configuration for invoices and accounting entries generated form POS orders.

cc\ @legalsylvain @elicoidal @pedrobaeza @Cedric-Pigeon

### CODE REVIEW

**Comment by adrienpeiffer** [comment]
Supersedes https://github.com/OCA/pos/pull/65

**Comment by legalsylvain** [comment]
:+1: Thanks.

File: pos_analytic_config/README.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    :alt: License: AGPL-3
+
+===================
+POS Analytic Config
+===================
+
+This module allows to use analytic account defined on POS configuration
+for invoices and accounting entries generated from POS orders.
+
+Usage
```
Comment (elicoidal) [inline pos_analytic_config/README.rst]: I miss a Configuration section with steps to set up and screenshot.

File: pos_analytic_config/README.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+===================
+POS Analytic Config
```
Comment (elicoidal) [inline pos_analytic_config/README.rst]: POS Analytic Account

File: pos_analytic_config/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline pos_analytic_config/models/__init__.py]: header

**Comment by elicoidal** [comment]
minor modifications. Otherwise LGTM (no test, no code review)

**Comment by pedrobaeza** [comment]
I still think that the module should be renamed to _pos_analytic_.

File: pos_analytic_config/README.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+This module allows to use analytic account defined on POS configuration
+for invoices and accounting entries generated from POS orders.
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/184/8.0
```
Comment (pedrobaeza) [inline pos_analytic_config/README.rst]: Change to 87 (this repo id)

**Comment by pedrobaeza** [comment]
Apart from this, the rest LGTM.

**Comment by elicoidal** [comment]
agree with pos_analytic

File: pos_analytic_config/models/__init__.py (hunk 3 above)
Comment (adrienpeiffer) [inline pos_analytic_config/models/__init__.py]: Done

**Comment by adrienpeiffer** [comment]
Thanks @elicoidal and @pedrobaeza !
Concerning the module name, I do not really agree. I agree with the PoV of @legalsylvain on https://github.com/OCA/pos/pull/65 . This is only a way to implement the analytic in the pos in the configuration of points of sales.

**Comment by pedrobaeza** [comment]
But my PoV is that you can join all these approaches in the same _pos_analytic_ module. And the end, you'll get a conflict of which one has the priority, so it's better to joint them in a module and solve this preference in it than splitting in a lot of mini-modules. We can put in "Known issues / Roadmap" that there can be other approaches for filling the analytic account.

**Comment by legalsylvain** [comment]
Ok we are all agree that there are many ways to implements analytic on PoS. Good news.

The question is one module per approach (mini modules) or one module for many approaches.

As the @adrienpeiffer module approach is the more generic (one analytic account for all PoS sales), the module could be indeed renamed into 'pos_analytic' and receive after other PR to make it more specific. (that is theory).

In practice, other way to implement analytic for PoS will require other extra modules. (account_analytic_default for example as mentionned on https://github.com/OCA/pos/pull/65).
- Extra dependencies + keep 1 module = install unnecessary modules; 
- Other solution : create many module : 
  -> pos_analytic_config
  -> pos_analytic_by_product ...

My 2 cents.

**Comment by elicoidal** [comment]
I am in favor of better granularity even if that means installing 2 modules.

**Comment by lmignon** [comment]
I'm also in favor of better granularity. +1 to keep `pos_analytic_config`

**Comment by sbidoul** [comment]
How about `pos_analytic_by_pos_config`?

**Comment by pedrobaeza** [comment]
I like the last name, but without repeating the word pos: `pos_analytic_by_config`

**Comment by adrienpeiffer** [comment]
Everyone agrees with `pos_analytic_by_config` ?

**Comment by sbidoul** [comment]
:+1: for pos_analytic_by_config

**Comment by adrienpeiffer** [comment]
@pedrobaeza @sbidoul @lmignon @elicoidal @legalsylvain @Cedric-Pigeon Now, it's `pos_analytic_by_config`. Thanks !

**Comment by pedrobaeza** [comment]
:+1:

**Comment by sbidoul** [comment]
I think this one can be merged.

**Comment by legalsylvain** [comment]
indeed.

**Pull request merged**